src/main/java/com/amazonaws/services/kinesis/connectors/KinesisConnectorRecordProcessor.java

```diff
@@ -126,8 +126,18 @@ public void processRecords(List<Record> records, IRecordProcessorCheckpointer ch
                 } else {
                     throw new RuntimeException("Transformer must implement ITransformer or ICollectionTransformer");
                 }
-            } catch (IOException e) {
+            } catch (Exception e) {
                 LOG.error(e);
+                try {
+                    final String lastSequenceNumberProcessed = buffer.getLastSequenceNumber();
```

Comment:
I have another consumer application for just archiving all the data (bad or good) in S3. So I'm not losing records. I hate the fact that the entire pipeline is stalled just because of some bad data.